### DISCUSSION TITLE
Fixed columns and rows, variable font size.

### DISCUSSION THREAD

**Discussion by loops** [wezterm/wezterm#3955] [Ideas]
I'm connecting to an old application that has a fixed size of 80 cols by 25 rows.   It would be nice to be able to change the window size and have the font automatically scale so that 80x25 is maintained, just with larger or smaller fonts as needed.

There is a terminal emulator named Kitty (https://github.com/cyd01/KiTTY), that is a fork of Putty, which has this feature, and it works very well.  But alas, even though it's cross-platform, that specific feature is Windows only, it doesn't work on Linux.

I was hoping maybe this could be accomplished with the current config system of Wezterm, but it appears the font handling can't be done by pixel sizes.

**Comment by wez** [comment]
The closest thing to that feature today is: https://wezfurlong.org/wezterm/config/lua/config/adjust_window_size_when_changing_font_size.html but that is coming at this from changing the font size, rather than changing the window size.

You could eg: resize the window larger, then use the IncreaseFontSize to make the font larger in that window.

Adjusting the font size to fit the window is a bit of an iterative process, as font sizes are not expressed in pixels, and we won't know the effective pixel dimensions without first trying a point size and testing it with the current font, dpi and scaling.  I'm worried about the latency impact that would have during eg: live window resizing, and I'm not sure if it is worth the engineering effort.

**Reply by loops** [reply]
Understood.  Thanks a lot for your reply.
